test/junit/scala/collection/IteratorTest.scala

```diff
@@ -213,6 +213,10 @@ class IteratorTest {
     assertEquals(it.drop(1).next(), it.drop(1).drop(1).next())
   }
 
+  @Test def dropDoesNotOverflow(): Unit = {
+    assertEquals(List(1, 2, 3).iterator.drop(1).drop(Int.MaxValue).drop(1).toList.isEmpty, true)
```

Comment:
List(1, 2, 3).iterator.drop(1).drop(Int.MaxValue) 
List(1, 2, 3).iterator.drop(Int.MaxValue).drop(1) 
List(1, 2, 3).reverseIterator.drop(Int.MaxValue).drop(1)
List(1, 2, 3).reverseIterator.drop(1).drop(Int.MaxValue).drop(1)
All those should be empty iterator, how about add more tests?

Reply:
Yeah the reverseIterator test is a good addition just in case.

Reply:
The last one doesn't seem relevant to Iterator though considering the appending is happening in the List side and when its passed to iterator its all the same

Reply:
Added requested tests